### PR TITLE
Add some tests for windows_path

### PR DESCRIPTION
**Pull request by smurawski**
Start a baseline of tests to validate the discussion around #111 (I've tested with 11.16.4 and 12.3.0)

/cc @pburkholder @sneal @btm @jdmundrawala @ksubrama @carpnick @spudnic @dawilliams @whoisjake

### CODE REVIEW

**Comment by jdmundrawala** [comment]
yes please

**Comment by lamont-granquist** [comment]
I think your integration test is only testing that chef-client itself picked up the updated path.  That does not check for the case in #111.  You need to setup a system path to a directory which has a runnable file in it.  Then in your integration test actually try to run that file and see if it succeeds.

**Comment by smurawski** [comment]
@lamont-granquist That was a poorly named test.  It's actually checking the registry where new processes get their environmental variables from, not just child processes.

File: test/integration/path/pester/minimal.path.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      ($Paths -contains 'C:\path_test_path') | should be $true
+    }
+
+    it "'C:\path_test_another_path' was added to the path" {
+      ($Paths -contains 'C:\path_test_another_path') | should be $true
+    }
+
+    it 'Child processes and shellouts have an updated path' {
+      'c:\paths.txt' | should contain 'C:\\path_test_another_path'
+      'c:\paths.txt' | should contain 'C:\\path_test_path'
```
Comment (lamont-granquist) [inline test/integration/path/pester/minimal.path.Tests.ps1]: stray `\\`'s here left over from old double quotes?

**Comment by lamont-granquist** [comment]
Yeah, but the registry is getting updated, and check on paths.txt is effectively just checking that chef-client updates its own path.  It doesn't check that an external powershell window really picks up the registry setting.  You should have the chef recipe drop a hello world script somewhere and then setup the windows_path to its directory and then have the powershell script try to invoke it through the path.

File: test/integration/path/pester/minimal.path.Tests.ps1 (hunk 1 above)
Comment (smurawski) [inline test/integration/path/pester/minimal.path.Tests.ps1]: the contain matcher is a regex.

**Comment by smurawski** [comment]
Ok, there are now tests that validate the updated path is being seen by newly launched external programs.  I use task scheduler to launch a new PowerShell session and log the path.

This fails on Server 2008 R2, but succeeds on Server 2012 and 2012 R2.

**Comment by ksubrama** [comment]
:+1:

**Pull request merged**